### PR TITLE
Reproduction case for Ruby 3.2 and kwargs detection

### PR DESCRIPTION
**Pull request by pirj**
Related to #1495

### CODE REVIEW

**Comment by pirj** [comment]
@eregon Do you happen to know what might have caused the failures on [3.2.0-preview2](https://github.com/rspec/rspec-mocks/actions/runs/3400302848/jobs/5654635455) and [ruby-head](https://github.com/rspec/rspec-mocks/actions/runs/3400302848/jobs/5654635939) while passed [just fine with 3.1](https://github.com/rspec/rspec-mocks/actions/runs/3400302848/jobs/5654635494)?

Is this still correct for Ruby 3.2?
```ruby

        if RSpec::Support::RubyFeatures.distincts_kw_args_from_positional_hash? && expected_args == actual_args
          expected_hash = expectation.expected_args.last
          actual_hash = args_for_multiple_calls.last.last
          if Hash === expected_hash && Hash === actual_hash &&
            (Hash.ruby2_keywords_hash?(expected_hash) != Hash.ruby2_keywords_hash?(actual_hash))
            actual_args += Hash.ruby2_keywords_hash?(actual_hash) ? " (keyword arguments)" : " (options hash)"
            expected_args += Hash.ruby2_keywords_hash?(expected_hash) ? " (keyword arguments)" : " (options hash)"
          end
        end
```


2.2 failure is unrelated, to be fixed by #1498

File: spec/rspec/mocks/matchers/receive_spec.rb [hunk 1]
```diff
@@ -109,6 +123,18 @@ module Mocks
...
+            it "expects to receive keyword args" do
+              dbl = instance_double(TestObject)
+              expect(dbl).to receive(:kw_args_method).with(a: 1, b: 2)
+              dbl.kw_args_method(a: 1, b: 2)
+            end
+
+            it "expects to receive keyword args with a hash" do
+              dbl = instance_double(TestObject)
+              expect(dbl).to receive(:kw_args_method).with({a: 1, b: 2})
+              dbl.kw_args_method(a: 1, b: 2)
```
Comment (eregon) [inline spec/rspec/mocks/matchers/receive_spec.rb:136]: Isn't this expected to fail? It expects a positional Hash but is given kwargs.
But I guess `expect(dbl).to receive` actually creates a method with no kwargs so then it should pass?

**Comment by eregon** [comment]
That code looks fine. I suspect it's a missing `ruby2_keywords` on the chain of calls, and so the flag is lost somewhere.
You can debug by printing `Hash.ruby2_keywords_hash?(h)` before the exception happens.

**Comment by eregon** [comment]
And 3.2 fixes a bug where some missing `ruby2_keywords` accidentally kept the flag: https://github.com/ruby/ruby/blob/master/NEWS.md

**Comment by pirj** [comment]
Thank you so much for the explanation, @eregon !

File: spec/rspec/mocks/matchers/receive_spec.rb (hunk 1 above)
Comment (pirj) [reply spec/rspec/mocks/matchers/receive_spec.rb:136]: This indeed should have failed on earlier Ruby versions, too.

**Comment by pirj** [comment]
Superseded by #1514

**Pull request closed without merging**